concepts/contexts.py

```diff
@@ -544,6 +544,11 @@ def properties(self):
     def bools(self):
         """list[tuple[bool, ...]]: Row-wise boolean relation matrix between objects and properties."""
         return self._intents.bools()
+    
+    @property
+    def shape(self):
+        """tuple[int, int]: Tuple of context dimensions."""
```

Comment:
Let's promote the first part into an actual type annotation.

I think the docstring should make unambiguous which number is which dimenstion :)

Reply:
Not sure what you mean, my original annotation was: `tuple[int, int]: Tuple of context dimensions (number of objects, number of properties).` Later modified to current version - agree needs to be changed. Do you mean something like `tuple[number of objects: int, number of properties: int]`?

Reply:
Sorry if I was unclear: Let's make this a proper type annotation, i.e. use the `typing` module and annotate the actual argument of the method instead of the docstring.

Cf. e.g.  https://github.com/xflr6/concepts/blob/5e785a0ada07ad06a1bcee5cd3fcf555d320f107/concepts/contexts.py#L514

It's probably worth to return a `typing.NamedTuple` for this then the output will be unambigous even without reading the docstring.

Reply:
I kept the classic tuple. We need some sutable names for `Context` dimensions If you think `typing.NamedTuple` is still worth it :)

Reply:
Sounds good, I think this is straightforward, WDYT?

```python
import typing

class ContextShape(typing.NamedTuple):
    """

    >>> ContextShape(403, 67)
    ContextShape(objects=403, properties=67)

    >>> n_objects, _ = ContextShape(403, 67)
    >>> nobjects
    403
    """"

    objects: int

    properties: int

    @property
    def rows(self) -> int:
        return self.objects

    @property
    def columns(self) -> int:
        return self.properties
```

Reply:
Thanks, module-level (not a nested class) SG.